@here/harp-mapview-decoder/lib/TilerService.ts

```diff
@@ -68,57 +60,32 @@ export class TilerService extends WorkerService {
     private async handleTileRequest(
         request: WorkerTilerProtocol.TileRequest
     ): Promise<WorkerServiceResponse> {
-        const tileIndex = this.m_tileIndexMap.get(request.index);
-        if (tileIndex === undefined) {
-            return { response: {} };
-        }
-
         const tileKey = TileKey.fromMortonCode(request.tileKey);
-        const tile = tileIndex.getTile(tileKey.level, tileKey.column, tileKey.row);
+        const tile = this.tiler.getTile(request.index, tileKey);
+
         return { response: tile || {} };
     }
 
     private async handleRegisterIndexRequest(
         message: WorkerTilerProtocol.RegisterIndexRequest
     ): Promise<WorkerServiceResponse> {
-        const tileIndex = this.m_tileIndexMap.get(message.id);
-        if (tileIndex === undefined) {
-            await this.writeIndex(message);
-        }
+        const geoJsonPath = typeof message.input === "string"
```

Comment:
It's not `geoJson` and it's not path `path`, it's `geoJson` or `url`, but in this case it would be best to just to stick with `input`, so we don't introduce artificial, misleading name.